File: src/test/java/com/hankcs/hanlp/corpus/MSR.java
/*
 * <author>Han He</author>
 * <email>me@hankcs.com</email>
 * <create-date>2018-06-24 10:34 AM</create-date>
 *
 * <copyright file="MSR.java">
 * Copyright (c) 2018, Han He. All Rights Reserved, http://www.hankcs.com/
 * This source is subject to Han He. Please contact Han He for more information.
 * </copyright>
 */
package com.hankcs.hanlp.corpus;

import com.hankcs.hanlp.corpus.io.IOUtil;
import com.hankcs.hanlp.utility.TestUtility;

/**
 * @author hankcs
 */
public class MSR
{
    public static String TRAIN_PATH = "data/test/icwb2-data/training/msr_training.utf8";
    public static String TEST_PATH = "data/test/icwb2-data/testing/msr_test.utf8";
    public static String GOLD_PATH = "data/test/icwb2-data/gold/msr_test_gold.utf8";
    public static String MODEL_PATH = "data/test/msr_cws";
    public static String OUTPUT_PATH = "data/test/msr_output.txt";
    public static String TRAIN_WORDS = "data/test/icwb2-data/gold/msr_training_words.utf8";
    public static String SIGHAN05_ROOT;

    static
    {
        SIGHAN05_ROOT = TestUtility.ensureTestData("icwb2-data", "http://sighan.cs.uchicago.edu/bakeoff2005/data/icwb2-data.zip");
        if (!IOUtil.isFileExisted(TRAIN_PATH))
        {
            System.err.println("请下载 http://sighan.cs.uchicago.edu/bakeoff2005/data/icwb2-data.zip 并解压为 data/test/icwb2-data");
            System.exit(1);
        }
    }
}

File: hanlp/datasets/cws/sighan2005/msr.py
# -*- coding:utf-8 -*-
# Author: hankcs
# Date: 2019-12-21 15:42
from hanlp.datasets.cws.sighan2005 import SIGHAN2005, make

SIGHAN2005_MSR_DICT = SIGHAN2005 + "#" + "gold/msr_training_words.utf8"
SIGHAN2005_MSR_TRAIN_FULL = SIGHAN2005 + "#" + "training/msr_training.utf8"
SIGHAN2005_MSR_TRAIN = SIGHAN2005 + "#" + "training/msr_training_90.txt"
SIGHAN2005_MSR_VALID = SIGHAN2005 + "#" + "training/msr_training_10.txt"
SIGHAN2005_MSR_TEST_INPUT = SIGHAN2005 + "#" + "testing/msr_test.utf8"
SIGHAN2005_MSR_TEST = SIGHAN2005 + "#" + "gold/msr_test_gold.utf8"

make(SIGHAN2005_MSR_TRAIN)
